### DISCUSSION TITLE
validateOnly resets all validation messages if a non existing field is validated

### DISCUSSION THREAD

**Discussion by gehrisandro** [livewire/livewire#3054] [7. Bugs (archived)]
### Description
Calling the validateOnly method for a non existing field (or an existing field without a rule), does currently reset the complete error bag.

### Exact steps to reproduce
1. Create component with a field and rule "foo".
2. Run validation with an invalid value for "foo".
3. Error bag has now a validation message for "foo".
4. Run validateOnly for "bar".
5. Error bag is now empty (in my opinion, the validation message for "foo" should persist)

### Stripped-down, copy-pastable code snippets
Regression test:
```php
/** @test */
public function validating_only_a_specific_field_will_preserve_other_validation_messages_if_the_field_doesnt_exist()
{
    $component = Livewire::test(ForValidation::class);

    $component
        ->set('foo', '')
        ->call('runValidation')
        ->assertSee('The foo field is required')
        ->call('runValidationOnlyWithFooRules', 'bar')
        ->assertSee('The foo field is required');
}
```

**Context**
- Livewire version: 2.4.1 (older versions have the same behaviour)

**Comment by gehrisandro** [comment]
Here the regression test mentioned above and possible fix: https://github.com/gehrisandro/livewire/commit/77f04cc53321af9589d4f483305d1f47ae5354a1

**Comment by nuernbergerA** [comment]
For me its intended behavior to reset the errorbar if I ONLY what to validate one attribute with `validateOnly()`

Should we throw an error if you validate an attribute that doesn't exist or doesn't have a rule?
@calebporzio thoughts on that?

**Comment by jehadassaf** [comment]
@nuernbergerA IMO this is not a feature request but rather a bug. One that I just encountered myself: when I forgot to add a validation rule for one of my wire models and I changed the property all the other errors were gone when they shouldn't have been.

Here's what the user story for it would look like:
Given I have a form with 10 wire model properties with validation rules for them
And another wire model property but with no validation rule for it
When I submit the form and I have errors for other fields in the form
And I change the field with no validation rule defined
I expect to retain the existing validation errors

**Comment by SimonMacIntyre** [comment]
I am experiencing this today. 

I use `validateOnly()` instead of a catch all to encapsulate different reactions to exceptions and different side effects I want to do upon a specific field being validated. I also manually add to the error bag for my convenience in the frontend. 

Now when I validate a different field, it clears my custom validation errors that should still be there. 

It is unintuitive in my opinion to assume that `validateOnly` means all previous error messages are now redundant. 

Now that I understand the bug (or, _feature_), I can bypass this issue by (inefficiently) `revalidating` everything, to allow the original custom error bag messages to re-propagate.
